uui-core/src/hooks/useVirtualList/useVirtualList.ts

```diff
@@ -53,6 +57,20 @@ export function useVirtualList<List extends HTMLElement = any, ScrollContainer e
         rowsSelector,
     ]);
 
+    useLayoutEffectSafeForSsr(() => {
+        if (scrollContainer.current?.clientHeight
+            && prevScrollContainerClientHeight
+            && scrollContainer.current?.clientHeight !== prevScrollContainerClientHeight
+        ) {
+            ++scrollContainerHeightChangesCount.current;
+        }
+        if (scrollContainerHeightChangesCount.current > 20 && !scrollContainerHeightIsNotLimited.current) {
```

Comment:
Probably we can just trigger on some unrealistically tall viewport? E.g. if we see that we'll get 10k pixels tall - warn, and stop attempting to handle more?